game/theater/frontline.py

```diff
@@ -65,10 +66,23 @@ def __init__(
         self.segments: List[FrontLineSegment] = [
             FrontLineSegment(a, b) for a, b in pairwise(route)
         ]
-        super().__init__(
-            f"Front line {blue_point}/{red_point}",
-            self.point_from_a(self._position_distance),
-        )
+        try:
+            super().__init__(
+                f"Front line {blue_point}/{red_point}",
+                self.point_from_a(self._position_distance),
```

Comment:
Isn't the bug in `_position_distance`? Why is that returning a point that isn't valid?

Reply:
Because the _position_distance calculation has a check for the minimum distance (_adjust_for_min_dist), which is based on the FRONTLINE_MIN_CP_DISTANCE constant. However, it does not check if the distance between the control points is smaller than this minimum distance.

Reply:
So `FRONTLINR_MIN_CP_DISTANCE` is wrong then?

Reply:
That's my hunch, but there are no comments or commit messages explaining why it has been set at 5000. There might be a legitimate reason to have it set higher than the capture radius. I was hesitant to change it just by myself, which is why I asked about it above.

Reply:
Everything below is from me reading through this section of code just now. I didn't know anything about how this worked before that, so take this with a grain of salt. I think the only person that is familiar with how front lines work is @walterroach, so maybe he can weigh in.

I think the problem is that `FRONTLINE_MIN_CP_DISTANCE` is being used although the actual limit that matters here is `FRONTLINE_MIN_PATH_LENGTH` because this isn't limiting the distance between two CPs, it's the minimum _path_ distance from the blue CP to the front line. It's apparently a requirement that frontline paths not be shorter than 5km, but that's not documented anywhere and so it seems this campaign doesn't obey that requirement.

There doesn't appear to be any real minimum for front line path length. We need enough space per segment to be able to deploy two lines of troops that are spaced reasonably and are not immediately in combat with each other. https://github.com/dcs-liberation/dcs_liberation/blob/697e2c2b11750ea2d080579563d88ee7c16382bf/gen/ground_forces/ai_ground_planner.py#L31-L41 shows the distances from the front line at which various unit types will be deployed. Tracing through https://github.com/dcs-liberation/dcs_liberation/blob/697e2c2b11750ea2d080579563d88ee7c16382bf/game/missiongenerator/flotgenerator.py#L741-L743, they are not limited by the length of the segment, so even if the front line is 1m from the end of the segment, the troops will just deploy behind the segment. In the event that behind the segment is water or something those troops will just not deploy, but imposing a minimum segment length would do nothing to change that.

I think the bug here is that `_adjust_for_min_dist` does not do what it claims:

https://github.com/dcs-liberation/dcs_liberation/blob/697e2c2b11750ea2d080579563d88ee7c16382bf/game/theater/frontline.py#L182-L186

That's what it did way back when front lines could only be straight lines. Ever since they've been segmented, what this actually does is ensure that the length of the path is at least 5km. I don't see any reason to do that and the minimum should just be removed entirely.

While it does make sense to avoid placing CPs too close together, there's no reason to check for that every time we compute front line data, and that's not what this does; it would be better checked during campaign generation (and actually compare CP distances rather than segment lengths). I don't think this is strictly necessary, but would be helpful to campaign designers. We don't currently do this, so doesn't need to be addressed in this PR.

Reply:
No responses from anyone that knows better, so I guess we'll just merge this and hope it's not making anything worse.